api/v1beta1/foundationdbcluster_types.go

```diff
@@ -70,6 +70,7 @@ func init() {
 // FoundationDBClusterSpec defines the desired state of a cluster.
 type FoundationDBClusterSpec struct {
 	// Version defines the version of FoundationDB the cluster should run.
+	// +kubebuilder:validation:Pattern:=^(\d+)\.(\d+)\.(\d+)$
```

Comment:
This is fine for now, but we may need to change it in the future to support pre-release versions with tags. When that comes we'll need to update the pattern both in this file and in the foundationdb_version.go file.

Reply:
That's fair 👍 I just think it's useful to provide as much validation as possible (or at least have the same validation in the code and the CRD) during the creation time otherwise the creation will fail and you have to dig through the logs or the events.